src/gui/tray/NotificationCache.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <set>
+#include <QObject>
+
+namespace OCC {
+
+struct Notification
+{
+    QString title;
+    QString message;
+};
+
+class NotificationCache : public QObject
+{
+    Q_OBJECT
+
+public:
+    explicit NotificationCache(QObject *parent = nullptr);
+
+    bool contains(const Notification &notification) const;
+
+    void insert(const Notification &notification);
+
+    void clear();
+
+private:
+    std::set<uint> _notifications;
+
+    uint calculateKey(const Notification &notification) const;
```

Comment:
nitpick: generally better to have the methods come first and then the variables in the private section

Reply:
I change it as we do it always that way, but in general I think it has benefits to have the data before the method. If I want to understand a piece of code, I always look at first at the data because the data somehow defines the class, and the goal of every operation that we do with the computer is to change some data.